src/cljdoc/server/pedestal.clj

```diff
@@ -104,43 +113,53 @@
                 :group/index    (index-pages/group-index (-> ctx :request :path-params) (::releases ctx))
                 :cljdoc/index   (index-pages/full-index (::releases ctx)))))])
 
+(defn seed-artifacts-keys
+  "An interceptor to prepare `::artifacts` key for [[artifact-data-loader]]."
+  [data-requirements]
+  (interceptor/interceptor
+   {:name ::seed-artifact-keys
+    :enter (fn seed-artifact-keys-inner [ctx]
+             (let [params (-> ctx :request :path-params)]
+               (assoc-in ctx [::artifacts (select-keys params [:group-id :artifact-id :version])] data-requirements)))}))
```

Comment:
The idea here is that this function would later on be extended to process `ctx` in more interesting ways, potentially resulting in different `::artifacts` values.

This would most likely be URL or query params that are provided through the `:request`. I wanted to keep this separate from the interceptor that's actually loading the data because this way the loading interceptor does not need to know about route names, where path or query params are stored and all that. Hope that makes sense.